### PR TITLE
fix(amazonq): Amazon Q chat should not mention indexing in progress when indexing finished

### PR DESCRIPTION
**Pull request by leigaol**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
Fix a bug when Amazon Q responds with still indexing message even when `@workspace` index is done

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt [hunk 1]
```diff
@@ -310,10 +310,11 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
     private fun sendMsgToLsp(msgType: LspMessage, request: String?): LspResponse {
         logger.info { "sending message: ${msgType.endpoint} to lsp on port ${encoderServer.port}" }
         val url = URL("http://localhost:${encoderServer.port}/${msgType.endpoint}")
-
+        // use 1h as timeout for index, 5 seconds for other APIs
+        val timeoutMs = if (msgType is LspMessage.Index) 3600*1000 else 5000
```
Comment (Will-ShaoHua) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt:314]: `60.minutes.inWholeMilliseconds`

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt [hunk 2]
```diff
@@ -310,10 +311,11 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
     private fun sendMsgToLsp(msgType: LspMessage, request: String?): LspResponse {
         logger.info { "sending message: ${msgType.endpoint} to lsp on port ${encoderServer.port}" }
         val url = URL("http://localhost:${encoderServer.port}/${msgType.endpoint}")
-
+        // use 1h as timeout for index, 5 seconds for other APIs
+        val timeoutMs = if (msgType is LspMessage.Index) 60.minutes.inWholeMilliseconds.toInt() else 5000
```
Comment (samgst-amazon) [inline plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt:315]: can/should these be enum?

Comment (leigaol) [reply plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextProvider.kt:315]: timeoutMs can only be integer.

**Review by rli — APPROVED** [review]

**Pull request merged**